cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1612,123 +1612,88 @@ data MultiSigScript = RequireSignature (Hash PaymentKey)
-             Just (String "all") -> case HMS.lookup "scripts" hm of
-                                      Just (Array allVectorValue) ->
-                                        Just . RequireAllOf $ gatherMultiSigScripts allVectorValue
-                                      _ -> Nothing
-             _ -> Nothing
-
-  atLeast :: HMS.HashMap Text Value -> Maybe MultiSigScript
-  atLeast hm =
-    case HMS.lookup "type" hm of
-      Just (String "atLeast") ->
-        case (HMS.lookup "required" hm, HMS.lookup "scripts" hm) of
-          (Nothing, Nothing) ->
-            error "atLeast multi-signature missing \"required\" and \"scripts\" keys"
-          (Just _, Nothing) ->
-            error "atLeast multi-signature missing \"required\" key"
-          (Nothing, Just _) ->
-            error "atLeast multi-signature missing \"scripts\" key"
-          (Just (Number sci), Just (Array listVectorValue)) ->
-             case toBoundedInteger sci of
-               Just reqInt ->
-                 Just $ fromJSONmOfnChecks
-                          (gatherMultiSigScripts listVectorValue)
-                          reqInt
-                          (Vector.length listVectorValue)
-               Nothing -> error $ "Error in multi-signature \"required\" key: "
-                                <> show sci <> " is not a valid Int"
-          (_, _) -> Nothing
-      _ -> Nothing
-
-
-  fromJSONmOfnChecks :: [MultiSigScript] -> Int -> Int -> MultiSigScript
-  fromJSONmOfnChecks keys required total
-    | total <= 0 = error $ "No payment key hashes were parsed from multiscript object: " <> show keys
-    | required <= 0 = error "The required number of payment key hashes cannot be less than or equal to 0."
-    | required == 1 = error "required is equal to one, you should use the \"any\" multi-signature script"
-    | length keys < required = error $ "required exceeds the number of payment key hashes. \
-                                       \Number of keys: " ++ show (length keys)
-                                     ++ " required: " ++ show required
-    | length keys == total = RequireMOf required keys
-    | otherwise = error $ "Cardano.Api.Typed.fromJSONmOfnChecks failure occured. required: "
-                        ++ show required ++ " total: " ++ show total
-                        ++ " keys: " ++ show keys
-
-  convertToHash :: Text -> Hash PaymentKey
-  convertToHash txt = case deserialiseFromRawBytesHex (AsHash AsPaymentKey) $ Text.encodeUtf8 txt of
-                        Just payKeyHash -> payKeyHash
-                        Nothing -> error $ "Error deserialising payment key hash: " <> Text.unpack txt
+  parseJSON = Aeson.withObject "MultiSigScript" $ \obj -> parseMultiSigScript obj
+
+convertValueToMultiSigScript :: Value -> Aeson.Parser MultiSigScript
+convertValueToMultiSigScript (Object valueHm) = parseMultiSigScript valueHm
+convertValueToMultiSigScript v = fail $ "Expected aeson object but got: " ++ show v
```

Comment:
This could also use `Aeson.withObject` and then we don't need a separate `convertValueToMultiSigScript`.